runtime/vm/jvminit.c

```diff
@@ -2844,6 +2844,23 @@ modifyDllLoadTable(J9JavaVM * vm, J9Pool* loadTable, J9VMInitArgs* j9vm_args)
 					return JNI_ERR;
 				}
 			}
+
+			/*
+			On Linux on Z it needs to open the libj9zlib to load libj9jit as it is used for AOT method
+			data compression which is currently only enabled on Z platform. To make sure when JVM loads 
+			libj9jit, libj9zlib is already loaded, we are openning it.
+			For more details see OpenJ9 PR#8561
+			*/
+			#if (defined(S390) && defined(LINUX))
+				char zlibDll[EsMaxPath];
```

Comment:
please introduce a scope as c89 requires that variables are defined at the top of a scope